presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java

```diff
@@ -495,24 +496,8 @@ public void unassignShard(long tableId, UUID shardUuid, String nodeIdentifier)
     @Override
     public Map<String, Long> getNodeBytes()
```

Comment:
No relevant to this specific change, but the shard ejector only considers shards that are actively referenced. There could be a condition where the node is hosting a lot more data (possibly a larger table was dropped but the clean up has not yet happened). In this case the shard ejector will clean up the shard from the old location and recovery to the new location will fail until the local clean up runs. 

Since the shard ejector is an effort to balance data across nodes for performance reasons, it would make sense to delay the ejection until there is space on the destination node.